### PR TITLE
Add BananaScan and EpsilonScan and rename FlameScansFR to LegacyScans

### LINKED ISSUE #16359: FlameScans.fr has changed to legacy-scans
### Source information

FlameScans.fr V1.4.25

### Source language

French

### Source new URL

https://legacy-scans.com/

### Other details

https://flamescans.fr/ has closed the website for DMCA problem but came back into https://legacy-scans.com/

thank you for this great app

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL is not updated yet.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #16638: EpsilonScan
### Source name

EpsilonScan

### Source link

https://epsilonscan.fr/

### Source language

French

### Other details

18+/NSFW = yes 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #16640: Banana-Scan
### Source name

Banana-Scan

### Source link

https://banana-scan.com/

### Source language

French

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by KirinRaikage**
Checklist:

Closes #16640 
Closes #16638 
Closes #16359 

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesiaGenerator.kt [hunk 1]
```diff
@@ -58,6 +59,7 @@ class MangaThemesiaGenerator : ThemeSourceGenerator {
         SingleLang("KumaPoi", "https://kumapoi.me", "id", isNsfw = true),
         SingleLang("Komiku.com", "https://komiku.com", "id", className = "KomikuCom"),
         SingleLang("Kuma Scans (Kuma Translation)", "https://kumascans.com", "en", className = "KumaScans", overrideVersionCode = 1),
+        SingleLang("Legacy Scans", "https://legacy-scans.com", "fr"),
```
Comment (alessandrojean) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesiaGenerator.kt:62]: Better to keep the old package name so users that have the existing one can update to the new one.

```suggestion
        SingleLang("Legacy Scans", "https://legacy-scans.com", "fr", packageName = "flamescansfr"),
```

File: multisrc/overrides/mangathemesia/legacyscans/src/LegacyScans.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package eu.kanade.tachiyomi.extension.fr.legacyscans
+
+import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class LegacyScans : MangaThemesia("Legacy Scans", "https://legacy-scans.com", "fr", dateFormat = SimpleDateFormat("MMMM d, yyyy", Locale.FRENCH))
```
Comment (alessandrojean) [inline multisrc/overrides/mangathemesia/legacyscans/src/LegacyScans.kt:7]:
```suggestion
class LegacyScans : MangaThemesia("Legacy Scans", "https://legacy-scans.com", "fr", dateFormat = SimpleDateFormat("MMMM d, yyyy", Locale.FRENCH)) {
    override val id = 8947802555328550956
}
```

Keep the old ID so users will not have to migrate after the update.

File: multisrc/overrides/mangathemesia/legacyscans/src/LegacyScans.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package eu.kanade.tachiyomi.extension.fr.legacyscans
```
Comment (alessandrojean) [inline multisrc/overrides/mangathemesia/legacyscans/src/LegacyScans.kt:1]:
```suggestion
package eu.kanade.tachiyomi.extension.fr.flamescansfr
```

Don't forget to rename the overrides folder from `legacyscans` back to `flamescansfr` as well.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

**Comment by KirinRaikage** [comment]
Thanks for reviewing!

**Review by alessandrojean — APPROVED** [review]

**Pull request merged**